packages/apps/kadena-docs/src/components/Header/useHeaderAnimation.tsx

```diff
@@ -0,0 +1,89 @@
+import { useRouter } from 'next/router';
+import { useCallback, useEffect, useRef } from 'react';
+
+interface IUseHeaderReturn {
+  listRef: React.RefObject<HTMLUListElement>;
+  backgroundRef: React.RefObject<HTMLDivElement>;
+}
+
+const getActiveItem = (
+  list: HTMLUListElement,
+  pathname: string,
+  clickedElement?: HTMLAnchorElement,
+): { item?: HTMLLIElement; idx: number } => {
+  let item = list.firstChild as HTMLLIElement;
+  let idx = 0;
+
+  Array.from(list.children).forEach((innerItem, innerIdx) => {
```

Comment:
nitpick: Right now you are looping through all of the children even after you have found the active item. You could opt to use Array.prototype.some() or Array.prototype.every() so that you can break the loop when you've found the active element

Reply:
I like it!